### DISCUSSION TITLE
X-show inside templates

### DISCUSSION THREAD

**Discussion by maximbashevoy** [alpinejs/alpine#3853] [1. Help] [answered]
I need to render the following structure (simplified):

```
Supplier 1
- Product 1
- Product 2
Supplier 2
- Product 3
...
```

Since I can't use templates inside of templates, I reduced the data to a single array of objects with different types ('supplier' and 'product'):

```
entries = [
   {id: 1, type: 'supplier', data: <supplier data>},
   {id: 2, type: 'product', data: <product data>},
]
```

which lets me get away with a single template like so:

```
<template x-for='entry in entries' :key='entry.id'>
    <div x-show='entry.type === "supplier"'>Render supplier</div>
    <div x-show='entry.type !== "supplier"'>Render product</div>
</template>
```

The problem is, only the first div ever evaluates. What am I missing?

**Comment by ekwoka** [comment, marked as answer]
You can use templates inside of templates.

Per the docs, Alpine needs the template to have only a single child.

**Comment by maximbashevoy** [comment]
Ah, got it. Thank you!

All I had to do was:

```
<template x-for='entry in entries' :key='entry.id'>
    <div>
        <div x-show='entry.type === "supplier"'>Render supplier</div>
        <div x-show='entry.type !== "supplier"'>Render product</div>
    </div>
</template>
```
